h/streamer/websocket.py

```diff
@@ -62,13 +63,17 @@ def __init__(self, sock, protocols=None, extensions=None, environ=None):
         self.registry = environ["h.ws.registry"]
 
         self._work_queue = environ["h.ws.streamer_work_queue"]
+        self.messages_received = 0
+        self.messages_sent = 0
```

Comment:
One way of doing these metrics without adding extra bits and bobs to the objects would to have some metrics gevent queues. Something like this:

```
from gevent import Queue
from collections import Counter

class Metrics:
    _queue = Queue(64000)
    
    @classmethod
    def add(cls, name, count=1)
        try:
            self._queue.add((name, count), block=False)
        except Full:
            # Oh well. You could compile this down if you wanted to at 
            # this point using `summary()` and re-adding the totals
            ...
            
    @classmethod
    def summary():
        # Pick off a finite amount so we don't keep going
        # as people add to it

        counter = Counter()
        for _ in range(self._queue.qsize()):
            try:
                key, value = queue.get(block=False)
                counter.update({key: value)
            except Empty:
                break
                
        return counter
```

The places where we might want to add a metric would then do:

```Metrics.add('ClientMessagesReceived')```

This has a few nice features:

 * Keeps the accounting out of the socket objects
 * Means we can globally report things and process them centrally
 * Always has a finite size and length